pytorch_translate/attention/dot_attention.py

```diff
@@ -27,36 +27,22 @@ def __init__(self, decoder_hidden_state_dim, context_dim, **kwargs):
         self.src_length_masking = kwargs.get("src_length_masking", True)
 
     def forward(self, decoder_state, source_hids, src_lengths):
+        # Reshape to bsz x src_len x context_dim
+        source_hids = source_hids.transpose(0, 1)
         # decoder_state: bsz x context_dim
         if self.input_proj is not None:
             decoder_state = self.input_proj(decoder_state)
-        # compute attention
-        attn_scores = (source_hids * decoder_state.unsqueeze(0)).sum(dim=2).t()
-
-        if self.src_length_masking:
-            max_src_len = source_hids.size()[0]
-            assert max_src_len == src_lengths.data.max()
```

Comment:
This assert got removed in the refactoring, is this intended?

Reply:
I'm a bit ambivalent about this one. It seems like this should already be taken care of upstream (eg. in the data loader).